packages/wow-ui/src/hooks/useFormControl.ts

```diff
@@ -0,0 +1,73 @@
+import type { ChangeEvent, FocusEvent } from "react";
+import { useLayoutEffect, useState } from "react";
+
+export type BaseVariantType = "default" | "typing" | "typed";
+
+interface FormControlProps<VariantType> {
+  defaultValue?: string;
+  value?: string;
+  maxLength?: number;
+  disabled?: boolean;
+  onChange?: (value: string) => void;
+  onBlur?: () => void;
+  onFocus?: () => void;
+  variant?: VariantType;
+}
+
+export function useFormControl<VariantType>({
+  defaultValue,
+  value: valueProp,
+  maxLength,
+  disabled,
+  onChange,
+  onBlur,
+  onFocus,
+}: FormControlProps<VariantType>) {
+  const [value, setValue] = useState(valueProp ?? defaultValue ?? "");
+  const [variant, setVariant] = useState<VariantType | BaseVariantType>(
+    "default"
+  );
+
```

Comment:
P3;
제어 컴포넌트로 조금 더 활용하기 위해서는 외부에서 새로운 `valueProp`이 들어와 props가 변경되었을때 이를 `value`라는 state에 반영해줄 수 있어야 할 것 같아요! 

```suggestion
  useEffect(() => {
    if (valueProp !== undefined) {
      setValue(valueProp);
    }
  }, [valueProp]);
```
그래야 요런 것도 가능할것 같습니다용
![Jul-24-2024 16-18-29](https://github.com/user-attachments/assets/0a42f852-dfeb-460a-8325-e7ab772c188c)

Reply:
비제어/제어 에 따라 비제어일 때는 내부 상태 관리 value 값으로, 제어일 때는 외부에서 전달받은 value 값으로 제어되도록 수정해두면서 요 부분도 해결되었습니다! [be6712e](https://github.com/GDSC-Hongik/wow-design-system/pull/83/commits/be6712eb95ee079a0d98a17ec73e03223c240975)

Reply:
좋아요옹 🥰